src/aggregationinfo.cpp

```diff
@@ -301,45 +301,53 @@ AggregationInfo AggregationInfo::SimpleMergeInfos(
 AggregationInfo AggregationInfo::SecureMergeInfos(
             std::vector<AggregationInfo> const &collidingInfosArg) {
     // Sort colliding Infos
-    std::vector<AggregationInfo> collidingInfos;
-    for (const AggregationInfo &info : collidingInfosArg) {
-        collidingInfos.push_back(info);
+    std::vector<size_t> sortedCollidingInfos(collidingInfosArg.size());
+    size_t pkCount = 0;
+    for (size_t i = 0; i < sortedCollidingInfos.size(); i++) {
+        sortedCollidingInfos[i] = i;
+        pkCount += collidingInfosArg[i].tree.size();
     }
     // Groups are sorted by message then pk then exponent
     // Each info object (and all of it's exponents) will be
     // exponentiated by one of the Ts
-    std::sort(begin(collidingInfos), end(collidingInfos));
-
-    std::vector<uint8_t*> sortedKeys;
-    for (auto &info : collidingInfos) {
-        for (auto &mapEntry : info.tree) {
-            sortedKeys.push_back(mapEntry.first);
+    std::sort(sortedCollidingInfos.begin(), sortedCollidingInfos.end(), [&collidingInfosArg](size_t a, size_t b) {
+        return collidingInfosArg[a] < collidingInfosArg[b];
+    });
+
+    std::vector<uint8_t*> msgAndPks;
+    std::vector<std::vector<uint8_t>> serPks;
+    std::vector<size_t> sortedKeys;
+    msgAndPks.reserve(pkCount);
+    serPks.reserve(pkCount);
+    sortedKeys.reserve(pkCount);
+    for (size_t i = 0; i < sortedCollidingInfos.size(); i++) {
+        for (auto &mapEntry : collidingInfosArg[sortedCollidingInfos[i]].tree) {
+            msgAndPks.emplace_back(mapEntry.first);
+            serPks.emplace_back(mapEntry.first + BLS::MESSAGE_HASH_LEN, mapEntry.first + BLS::MESSAGE_HASH_LEN + BLSPublicKey::PUBLIC_KEY_SIZE);
+            sortedKeys.emplace_back(sortedKeys.size());
         }
     }
     // Pks are sorted by message then pk
-    std::sort(begin(sortedKeys), end(sortedKeys), BLSUtil::BytesCompare80());
```

Comment:
Are we using BytesCompare80 anymore? If not we can remove it from BLSUtil

Reply:
still used in SortIntoVectors